devito/finite_differences/derivative.py

```diff
@@ -141,6 +149,24 @@ def __new__(cls, expr, *dims, **kwargs):
         obj._x0 = kwargs.get('x0', None)
         return obj
 
+    def __call__(self, x0=None, fd_order=None, side=None):
```

Comment:
this will require some engineering to make it **super quick**

Reply:
e.g., use try-catch instead of isinstance below

however, this will need some profiling with cProfile I suspect

Reply:
>this will require some engineering to make it super quick
This is done a single time, so there won't be much of a hit, and it's a single simple change